pkg/source/firefly/config.go

```diff
@@ -0,0 +1,156 @@
+// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firefly
+
+import (
+	"os"
+
+	"github.com/spf13/pflag"
+	"go.uber.org/zap"
+
+	"go.thethings.network/lorawan-stack-migrate/pkg/source"
+	"go.thethings.network/lorawan-stack-migrate/pkg/source/firefly/client"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+type Config struct {
+	client.Config
+	src source.Config
+
+	appID           string
+	frequencyPlanID string
+	joinEUI         string
+	macVersion      string
+	invalidateKeys  bool
+
+	derivedMacVersion ttnpb.MACVersion
+	derivedPhyVersion ttnpb.PHYVersion
+
+	flags *pflag.FlagSet
+}
+
+var logger *zap.SugaredLogger
+
+// New returns a new Firefly configuration.
+func New() *Config {
```

Comment:
I think the name is confusing looking from the package perspective since I'd think `firefly.New()` would result in a new Firefly Source, not configuration. In other sources the configuration is in a subpackage to where `config.New()` is self-explanatory. I'd change this to `NewConfig` or move configuration to a `config` subpackage, but I'd ask @adriansmares to give his opinion first.